README.md

```diff
@@ -1,46 +1,92 @@
 # vrops-exporter
 Prometheus exporter to scrape metrics from vRealize Operations Manager
 
-## Running the exporter
+### Running the exporter
 
-The container is defaulting to /vrops-exporter path. Use vrops-exporter.py with one of the two ways of specifying the necessary credentials. They could also be mixed of course.
+The container is defaulting to /vrops-exporter path. 
+Use exporter.py with one of the two ways of specifying the necessary credentials. 
+They could also be mixed of course.
 
 1. CLI
 
-```
-Usage: vrops-exporter.py [options]
-```
-Options:
-
-short | long | description
---- | --- | ---
-  -h | --help |           show this help message and exit
-  -u USER | --user=USER | specify user to log in
-  -p PASSWORD | --password=PASSWORD | specify password to log in
-  -o PORT | --port=PORT | specify exporter port
-  -d | --debug    |       enable debug
+    ```
+    Usage: pyhton3 exporter.py [options]
+    ```
+    Options:
+    
+    short | long | description
+    --- | --- | ---
+      -h | --help |           show this help message and exit
+      -u USER | --user=USER | specify user to log in
+      -p PASSWORD | --password=PASSWORD | specify password to log in
+      -o PORT | --port=PORT | specify exporter port
+      -a | --atlas | specify atlas configfile 
+      -d | --debug    |       enable debug
 
 
 2. Enviroment variables
 
-```
-USER
-PASSWORD
-PORT
-```
+    ```
+    USER
+    PASSWORD
+    PORT
+    ```
+### Architecture
+
+![](images/architecture.jpeg)
+
+### How to develop a collector
 
-## Service discovery way of querying
-Query with: ``http://localhost:1234/?target=yourDNSName``
+1. [optional] Provide a simple configfile that contains at least one resource in the structure shown below to the main directory. 
 
+    ```json
+    [
+       {
+          "labels": {
+                   "job": "vrops",
+                   "server_name": "VROPS_DNS_NAME" 
+          }
+       }
+    ]
+    ```
+2. Add the desired *statkeys* that the collector should collect to `statkey.yaml`. This is where pairs of a *statkey* 
+and a *label* are described. The *statkey* follows VMWARE notation and the *label* follows best practices as it should appear in prometheus.
+    
+    statkey:
+    [VMWARE Documentation | Metrics for vCenter Server Components](https://docs.vmware.com/en/vRealize-Operations-Manager/7.5/com.vmware.vcom.metrics.doc/GUID-9DB18E49-5E00-4534-B5FF-6276948D5A09.html)
+    
+    label:
+    [Prometheus | Metric and label naming](https://prometheus.io/docs/practices/naming/)
+    
+    ```yaml
+    HostSystemCollector:
+       - label: "memory_usage_average_percentage"
+         statkey: "mem|usage_average"
+       - label: "summary_running_VMs_number"
+         statkey: "summary|number_running_vms"
+    ```
```

Comment:
little hint, that there will be another file for getting properties, which is not yet in place.